runtime/src/snapshot_utils.rs

```diff
@@ -1598,6 +1613,74 @@ pub fn bank_from_latest_snapshot_archives(
     ))
 }
 
+/// Build bank from a snapshot (a snapshot directory, not a snapshot archive)
+#[allow(clippy::too_many_arguments)]
+pub fn bank_from_snapshot_dir(
+    account_paths: &[PathBuf],
+    bank_snapshot: &BankSnapshotInfo,
+    genesis_config: &GenesisConfig,
+    runtime_config: &RuntimeConfig,
+    debug_keys: Option<Arc<HashSet<Pubkey>>>,
+    additional_builtins: Option<&Builtins>,
+    account_secondary_indexes: AccountSecondaryIndexes,
+    limit_load_slot_count_from_snapshot: Option<usize>,
+    shrink_ratio: AccountShrinkThreshold,
+    verify_index: bool,
+    accounts_db_config: Option<AccountsDbConfig>,
+    accounts_update_notifier: Option<AccountsUpdateNotifier>,
+    exit: &Arc<AtomicBool>,
+) -> Result<(Bank, BankFromArchiveTimings)> {
+    // next_append_vec_id is not used when loading from a snapshot directory.  It is kept because the from_dir
+    // and from_archive paths share many function calls, and it is easier to keep the same function signatures.
+    // It is initialized to 1 to avoid substraction underflow in reconstruct_accountsdb_from_fields.  If it is
+    // initialized to 0, it is never incremented to non-zero, so the substraction underflow will occur, causing
+    // test_bank_from_snapshot_dir to fail.
+    let next_append_vec_id = Arc::new(AtomicAppendVecId::new(1));
+
+    let measure_build_storage = Measure::start("build storage from snapshot dir");
+    let storage =
+        build_storage_from_snapshot_dir(bank_snapshot, account_paths, next_append_vec_id.clone())?;
+    info!("{}", measure_build_storage);
+
+    let next_append_vec_id =
+        Arc::try_unwrap(next_append_vec_id).expect("this is the only strong reference");
+    let storage_and_next_append_vec_id = StorageAndNextAppendVecId {
+        storage,
+        next_append_vec_id,
+    };
+    let mut measure_rebuild = Measure::start("rebuild bank from snapshots");
+    let bank = rebuild_bank_from_snapshot(
+        bank_snapshot,
+        account_paths,
+        storage_and_next_append_vec_id,
+        genesis_config,
+        runtime_config,
+        debug_keys,
+        additional_builtins,
+        account_secondary_indexes,
+        limit_load_slot_count_from_snapshot,
+        shrink_ratio,
+        verify_index,
+        accounts_db_config,
+        accounts_update_notifier,
+        exit,
+    )?;
+    measure_rebuild.stop();
+    info!("{}", measure_rebuild);
+
+    // Skip bank.verify_snapshot_bank.  Subsequent snapshot requests/accounts hash verification requests
+    // will calculate and check the accounts hash, so we will still have safety/correctness there.
+    bank.set_initial_accounts_hash_verification_completed();
+
+    let timings = BankFromArchiveTimings {
+        rebuild_bank_from_snapshots_us: measure_rebuild.as_us(),
+        full_snapshot_untar_us: measure_build_storage.as_us(),
+        incremental_snapshot_untar_us: 0,
```

Comment:
Don't do it in this PR - but we should rename this field. We're not untaring anything here. This is just rebuild time.